solr/core/src/java/org/apache/solr/response/transform/ChildDocTransformer.java

```diff
@@ -126,6 +128,12 @@ public void transform(SolrDocument rootDoc, int rootDocId) {
           continue;
         }
 
+        // check whether doc is "live"
+        if (liveDocs != null && !liveDocs.get(docId)) {
```

Comment:
```suggestion
        if (liveDocs != null && !liveDocs.get(docId - segBaseId)) {
```
See if your test could have caught this by adding a doc & committing up front, thus creating an additional segment?  You could wrap in a random().nextBoolean()

Reply:
Oh oops, my bad.
Fixed and added your suggestion to the test.